### DISCUSSION TITLE
[Art competition] Return-oriented Programming

### DISCUSSION THREAD

**Discussion by jchaemin** [prosyslab-classroom/cs348-information-security#407] [Art competition]
![DALL·E 2023-05-24 17 41 49 - A tiny boy with infinite stars in his heart looking up to a shining star in a dark sky, oil painting](https://github.com/prosyslab-classroom/cs348-information-security/assets/105200949/7311eb88-ec1c-4abe-bb80-9e2f5547e94e)

## Short Description
For me, [Return-oriented Programming] means [A tiny boy with infinite stars surrounding him looking up to a shining star in a dark sky, oil painting]".

## Long Description (up to 3 sentences)
The large, shiny star represents the boy's dream while the tiny stars surrounding the boy represent gadgets that are already available to the boy. Even though the shining star in the sky may seem unreachable at this point, the boy will later realize that the tiny stars needed to build a large shining star was already inside him. ROP can be summarized with the following quote: "Everything I need is already inside me."

**Comment by KihongHeo** [comment]
Great. Looks like a [little prince](https://kawarthanow.com/2022/11/22/beloved-tale-of-the-little-prince-comes-to-the-peterborough-theatre-guild-stage-in-december/).
